### PR TITLE
Experiments logging

### PR DESCRIPTION
**Pull request by tomer-algo**
- upload images to wandb, with joints as mask
- add experiment for sms (weight 0 for non-relevant joints)

### CODE REVIEW

File: lib/dataset/nanit_mpii.py [hunk 1]
```diff
@@ -76,23 +79,31 @@ def handle_validation_dataset(self, cfg):
...
-                d['headbbox'] = transform_box_xywh(d['headbbox'], rotation_matrix)
-                joints_vis_arr = np.array(d['joints_vis']) == 1
-                joints_arr = np.array(d['joints'])
-                joints_for_transformation = joints_arr[joints_vis_arr].reshape(-1, 1, 2)
+                d['headbbox'] = transform_box_xywh(d['headbbox'], rotation_matrix)                 # keep metadata updated
+                joints_for_transformation = joints_arr[joints_vis_arr == 1].reshape(-1, 1, 2)
                 joints_transformed = cv2.transform(joints_for_transformation, rotation_matrix)
-                joints_arr[joints_vis_arr] = joints_transformed[:, 0, :]
-                d['joints'] = [list(p) for p in joints_arr]
+                joints_arr[joints_vis_arr == 1] = joints_transformed[:, 0, :]
```
Comment (felixNan) [inline lib/dataset/nanit_mpii.py:100]: Why joints_vis_arr is 0 and 1 and not True and False?

File: lib/dataset/nanit_mpii.py [hunk 2]
```diff
@@ -115,28 +126,26 @@ def _get_db(self):
...
-                joints[:, 0:2] = joints[:, 0:2] - 1
                 joints_vis = np.array(a['joints_vis'])
                 assert len(joints) == self.num_joints, \
                     'joint num diff: {} vs {}'.format(len(joints),
                                                       self.num_joints)
 
+                # Set joints with zero weight as not visible
+                joints[self.joints_with_zero_weight, 0:2] = -1
+                joints_vis[self.joints_with_zero_weight] = 0
+
```
Comment (felixNan) [inline lib/dataset/nanit_mpii.py:141]: We need to sit on this part.

File: lib/nanit_utils/utils_functions.py [hunk 3]
```diff
@@ -28,3 +28,47 @@ def transform_box_xywh(box_xywh, transformation_matrix):
...
+            cv2.circle(mask_gt, (int(joint_gt[0]), int(joint_gt[1])), 3, joint_label_index, -1)
+            cv2.circle(mask_pred, (int(joint_pred[0]), int(joint_pred[1])), 3, joint_label_index, -1)
+
+        image_np = images_np[i]
+        curr_image_to_upload = wandb.Image(image_np,
+                                           masks={'predictions': {'mask_data': mask_pred, 'class_labels': joints_labels_wandb},
+                                                  'ground_truth': {'mask_data': mask_gt, 'class_labels': joints_labels_wandb}
+                                                 }
+                                           )
+        images_to_upload_list.append(curr_image_to_upload)
```
Comment (felixNan) [inline lib/nanit_utils/utils_functions.py:71]: Consider putting in a separate function.

File: lib/dataset/nanit_mpii.py (hunk 2 above)
Comment (tomer-algo) [reply lib/dataset/nanit_mpii.py:141]: to mark joint as not visible
- the joint coordinates are set to -1
- joints vis set to 0

File: lib/nanit_utils/utils_functions.py (hunk 3 above)
Comment (tomer-algo) [reply lib/nanit_utils/utils_functions.py]: add a function

File: lib/dataset/nanit_mpii.py (hunk 1 above)
Comment (tomer-algo) [reply lib/dataset/nanit_mpii.py]: it saved as 0/1 in json file (under 'joints_vis')
and self.jnt_visible is also expected to be 0/1

added - do transform only if at least one joint is visible

File: lib/dataset/nanit_mpii.py (hunk 2 above)
Comment (felixNan) [reply lib/dataset/nanit_mpii.py:142]: You have a similar code up above and I wonder why do you need it.

**Review by felixNan — APPROVED** [review]

**Pull request merged**